app/src/test/java/com/quran/labs/androidquran/presenter/QuranImportPresenterTest.kt

```diff
@@ -0,0 +1,109 @@
+package com.quran.labs.androidquran.presenter
+
+import android.content.ContentResolver
+import android.content.Context
+import android.net.Uri
+import android.os.ParcelFileDescriptor
+
+import com.quran.labs.androidquran.model.bookmark.BookmarkImportExportModel
+import com.quran.labs.androidquran.model.bookmark.BookmarkModel
+
+import org.junit.Before
+import org.junit.Test
+
+import java.io.ByteArrayInputStream
+import java.io.FileNotFoundException
+import java.io.InputStream
+
+import io.reactivex.observers.TestObserver
+import okio.BufferedSource
+
+import com.google.common.truth.Truth.assertThat
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+
+class QuranImportPresenterTest {
+
+  private lateinit var appContext: Context
+  private lateinit var presenter: QuranImportPresenter
+
+  @Before
+  fun setup() {
+    appContext = mock(Context::class.java)
+    val model = mock(BookmarkImportExportModel::class.java)
+    presenter = QuranImportPresenter(appContext, model, mock(BookmarkModel::class.java))
+  }
+
+  @Test
+  @Throws(FileNotFoundException::class)
+  fun testParseExternalFile() {
+    val `is`: InputStream = ByteArrayInputStream(ByteArray(32))
+    val resolver = mock(ContentResolver::class.java)
+    `when`(resolver.openInputStream(any())).thenReturn(`is`)
```

Comment:
nit - instead do something like

```
import org.mockito.Mockito.`when` as whenever
```

and then you can use `whenever`

Reply:
done in new PR 👍🏻